### PR TITLE
Fail MD5 validation if server MD5 is missing

### PR DESCRIPTION
**Pull request by tjcelaya**
#298

Updating this code path so we cover situations where we got back a response that is somehow missing an MD5. Minor detour while attempting to remove the `entity == null ` early return:

It's currently possible to call `put("/path", null)` and get a response. Some extra testing revealed that enabling client-side encryption and sending a null string causes the following exception:

```
java.lang.NullPointerException
	at com.joyent.manta.client.crypto.EncryptingEntity.<init>(EncryptingEntity.java:98)
	at com.joyent.manta.http.EncryptionHttpHelper.httpPut(EncryptionHttpHelper.java:189)
	at com.joyent.manta.client.MantaClient.put(MantaClient.java:1084)
	at com.joyent.manta.client.MantaClient.put(MantaClient.java:1134)
	at com.joyent.manta.client.MantaClientPutIT.testPutWithStringUTF8(MantaClientPutIT.java:79)
```

It seems like the simplest path forward is to disallow null strings from being sent as content (empty strings are fine).

### CODE REVIEW

**Comment by tjcelaya** [comment]
@dekobon @cburroughs whenever you have time please check this PR

File: java-manta-client/src/test/java/com/joyent/manta/http/StandardHttpHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        final StandardHttpHelper helper = new StandardHttpHelper(connCtx, connectionFactory, config);
+
+        // it's the default but let's just be explicit
+        config.setVerifyUploads(true);
+
+        Assert.assertThrows(MantaChecksumFailedException.class, () ->
+                helper.httpPut("/path", null, NoContentEntity.INSTANCE, null));
+    }
+
+}
```
Comment (cburroughs) [inline java-manta-client/src/test/java/com/joyent/manta/http/StandardHttpHelperTest.java]: nit:newline

File: java-manta-client/src/main/java/com/joyent/manta/http/StandardHttpHelper.java [hunk 2]
```diff
@@ -351,13 +351,11 @@ protected static void validateChecksum(final DigestedEntity entity,
...
             throws MantaChecksumFailedException {
-        if (entity == null) {
-            return;
-        }
+        Validate.notNull(entity, "Request body required");
 
         if (serverMd5 == null || serverMd5.length == 0) {
-            LOGGER.warn("No cryptographic check performed by the server");
-            return;
+            final String msg = "Server calculated MD5 is missing";
```
Comment (cburroughs) [inline java-manta-client/src/main/java/com/joyent/manta/http/StandardHttpHelper.java]: This is a separate path from MANTA-3368, correct?

File: java-manta-client/src/test/java/com/joyent/manta/http/StandardHttpHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+import java.util.Base64;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by tomascelaya on 8/11/17.
```
Comment (cburroughs) [inline java-manta-client/src/test/java/com/joyent/manta/http/StandardHttpHelperTest.java]: Did you intend to leave this from the ide?

File: java-manta-client/src/main/java/com/joyent/manta/http/StandardHttpHelper.java (hunk 2 above)
Comment (tjcelaya) [inline java-manta-client/src/main/java/com/joyent/manta/http/StandardHttpHelper.java]: Correct. This method is only called on standard PUT requests while that ticket is about MPU commit.

File: java-manta-client/src/test/java/com/joyent/manta/http/StandardHttpHelperTest.java (hunk 3 above)
Comment (tjcelaya) [inline java-manta-client/src/test/java/com/joyent/manta/http/StandardHttpHelperTest.java]: No, removed it from the template so it won't show up again.
